src/Microsoft.DotNet.Cli.Utils/ProjectContextExtensions.cs

```diff
@@ -63,5 +69,118 @@ public static string GetDefaultRootOutputPath(ProjectContext context, string cur
 
             return rootOutputPath;
         }
+        
+        internal static void MakeRunnable(this ProjectContext context, string outputPath, string configuration)
+        {
+            context
+                .ProjectFile
+                .Files
+                .GetCopyToOutputFiles()
+                .StructuredCopyTo(context.ProjectDirectory, outputPath)
+                .RemoveAttribute(FileAttributes.ReadOnly);
+
+            var exporter = context.CreateExporter(configuration);
+
+            if (context.TargetFramework.IsDesktop())
+            {
+                exporter
+                    .GetDependencies()
+                    .SelectMany(e => e.RuntimeAssets())
+                    .CopyTo(outputPath);
+
+                GenerateBindingRedirects(context, outputPath, configuration);
+            }
+            else
+            {
+                exporter
+                    .GetDependencies(LibraryType.Package)
+                    .WriteDepsTo(Path.Combine(outputPath, context.ProjectFile.Name + FileNameSuffixes.Deps));
+
+                exporter.GetDependencies(LibraryType.Project)
+                    .SelectMany(e => e.RuntimeAssets())
```

Comment:
Why do we copy runtime assets here? Shouldn't this already be happening in publish?